int/apihandler/cmd/execute_function.go

```diff
@@ -1,46 +1,54 @@
-package apihandler
+package cmd
 
 import (
+	"sync"
+
 	"github.com/go-openapi/runtime/middleware"
+	"github.com/massalabs/thyra/api/swagger/server/models"
 	"github.com/massalabs/thyra/api/swagger/server/restapi/operations"
 	"github.com/massalabs/thyra/pkg/node"
 	"github.com/massalabs/thyra/pkg/node/base58"
 	sendOperation "github.com/massalabs/thyra/pkg/node/sendoperation"
 	callSC "github.com/massalabs/thyra/pkg/node/sendoperation/callsc"
+	"github.com/massalabs/thyra/pkg/wallet"
 )
 
-const (
-	errorCodeSendOperation = "Execute-0001"
-	errorCodeUnknownKeyID  = "Execute-0002"
-)
+func NewExecuteFunction(walletStorage *sync.Map) operations.CmdExecuteFunctionHandler {
+	return &functionExecuter{walletStorage: walletStorage}
+}
 
-func ExecuteFunction(params operations.CmdExecuteFunctionParams) middleware.Responder {
+type functionExecuter struct {
+	walletStorage *sync.Map
+}
+
+func (f *functionExecuter) Handle(params operations.CmdExecuteFunctionParams) middleware.Responder {
 	addr, err := base58.CheckDecode((*params.Body.At)[1:])
 	if err != nil {
 		panic(err)
 	}
 
 	addr = addr[1:]
 
-	if *params.Body.KeyID != "default" {
+	value, ok := f.walletStorage.Load(*params.Body.KeyID)
+	if !ok {
 		e := errorCodeUnknownKeyID
-		msg := "Error: unknown key id (" + *params.Body.KeyID + ")"
+		msg := "Error: unknown wallet id: " + *params.Body.KeyID
+
 		return operations.NewCmdExecuteFunctionUnprocessableEntity().WithPayload(
-			&operations.CmdExecuteFunctionUnprocessableEntityBody{
+			&models.Error{
 				Code:    &e,
 				Message: &msg,
 			})
 	}
 
-	pubKey, err := base58.CheckDecode("zkTGqfwJp43tY4FPgRXC7fr2xML3kDQ8bch15SpnDehuxWiKS")
-	if err != nil {
-		panic(err)
+	storedWallet, ok := value.(*wallet.Wallet)
+	if !ok {
+		panic("stored value is not a wallet")
```

Comment:
We can leave it like that, but we should create a new issue specifying that the panic should be removed.